test-data/unit/check-based-infer-function-types.test

```diff
@@ -267,3 +269,101 @@ class S:
     def f(self, f: int) -> str: ...
     @overload
     def f(self, f: str) -> int: ...
+
+
+[case testCheckUntypedDefs]
+# flags: --no-check-untyped-defs --allow-untyped-defs
+
+def foo(a=1):
+    1 + ""
```

Comment:
because of `--no-check-untyped-defs --allow-untyped-defs` this function is not typed. perhaps this function could be inferred, but it's body isn't checked. That might be a good idea, but for another day.